File: 프로그래머스/level1_소수 만들기.py
def is_prime(x):
    for i in range(2, int(x ** 0.5) + 1):
        if x % i == 0:
            return False
    return True

def solution(nums):
    answer = 0
    for i in range(len(nums) - 2):
        for j in range(i + 1, len(nums) - 1):
            for k in range(j + 1, len(nums)):
                if is_prime(nums[i] + nums[j] + nums[k]):
                    answer += 1

    return answer

print(solution([1, 2, 7, 6, 4]))

File: 프로그래머스 복습(java)/src/level1/level1_소수_만들기.java
package level1;

public class level1_소수_만들기 {
    static class Solution {

        public boolean isPrime(int num) {
            for (int i = 2; i < Math.sqrt(num) + 1; i++) {
                if (num % i == 0) {
                    return false;
                }
            }
            return true;
        }
        public int solution(int[] nums) {
            int answer = 0;
            for (int i = 0; i < nums.length - 2; i++) {
                for (int j = i + 1; j < nums.length - 1; j++) {
                    for (int k = j + 1; k < nums.length; k++) {
                        int sum_value = nums[i] + nums[j] + nums[k];
                        if (isPrime(sum_value)) {
                            answer++;
                        }
                    }
                }
            }
            return answer;
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums1 = {1, 2, 3, 4}, nums2 = {1, 2, 7, 6, 4};
        solution.solution(nums1);
        solution.solution(nums2);
    }
}
